
From: Pavel Machek <pavel@ucw.cz>

Currently, when both S4 and S4bios are available, swsusp always chooses
S4bios and makes S4 unavailable.  Bad idea as S4bios needs completely
different setup.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/acpi/sleep/main.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletion(-)

diff -puN drivers/acpi/sleep/main.c~swsusp-do-not-disable-platform-swsusp-because-s4bios-is-available drivers/acpi/sleep/main.c
--- 25/drivers/acpi/sleep/main.c~swsusp-do-not-disable-platform-swsusp-because-s4bios-is-available	Tue Aug 17 15:15:34 2004
+++ 25-akpm/drivers/acpi/sleep/main.c	Tue Aug 17 15:15:34 2004
@@ -217,7 +217,8 @@ static int __init acpi_sleep_init(void)
 				sleep_states[i] = 1;
 				printk(" S4bios");
 				acpi_pm_ops.pm_disk_mode = PM_DISK_FIRMWARE;
-			} else if (sleep_states[i])
+			}
+			if (sleep_states[i])
 				acpi_pm_ops.pm_disk_mode = PM_DISK_PLATFORM;
 		}
 	}
_
